etcdserver/server.go

```diff
@@ -184,6 +186,14 @@ type EtcdServer struct {
 	forceVersionC chan struct{}
 
 	msgSnapC chan raftpb.Message
+
+	cpMu sync.Mutex // guards compactionPaused
+	// When sending a snapshot, etcd will pause compaction.
+	// After receives a snapshot, the slow follower needs to get all the entries right after
+	// the snapshot sent to catch up. If we do not pause compaction, the log entries right after
+	// the snapshot sent might already be compacted. It happens when the snapshot takes long time
+	// to send and save. Pausing compaction avoids triggering a snapshot sending cycle.
+	compactionPaused bool
```

Comment:
`compactionPaused` is really tracking whether a snapshot is being sent. Something like `isSendingSnapshot` would probably be a bit clearer since paused <=> sending at the moment (e.g., the sender just has to say it's sending, instead of worrying about whether it should pause compaction).

Is it possible to get another message on msgSnapC while another snapshot is getting sent out? If that's the case, the mutex stuff can be nuked for an atomic int (`inflightSnapshots`?).